dsl/src/main/scala/asobu/dsl/extractors/AuthInfoExtractorBuilder.scala

```diff
@@ -1,20 +1,42 @@
 package asobu.dsl.extractors
 
-import cats.data.Xor
-import asobu.dsl.Extractor
+import cats.data.Kleisli
+import asobu.dsl.{ExtractResult, RequestExtractor}
+import ExtractResult._
+import cats.sequence.RecordSequencer
 import play.api.mvc.RequestHeader
 import play.api.mvc.Results._
-import shapeless.HList
-
+import shapeless._
+import shapeless.ops.record.Selector; import record._
 import scala.concurrent.Future
+import asobu.dsl.CatsInstances._
+import concurrent.ExecutionContext.Implicits.global
 
 class AuthInfoExtractorBuilder[AuthInfoT](buildAuthInfo: RequestHeader ⇒ Future[Either[String, AuthInfoT]]) {
-  import scala.concurrent.ExecutionContext.Implicits.global
 
-  def apply[Repr <: HList](toRecord: AuthInfoT ⇒ Repr): Extractor[Repr] = { req ⇒
-    buildAuthInfo(req).map {
-      case Left(msg)       ⇒ Xor.Left(Unauthorized(msg))
-      case Right(authInfo) ⇒ Xor.Right(toRecord(authInfo))
+  def apply[Repr <: HList](toRecord: AuthInfoT ⇒ Repr): RequestExtractor[Repr] =
+    apply.map(toRecord)
+
+  def apply(): RequestExtractor[AuthInfoT] = Kleisli(
+    buildAuthInfo.andThen(_.map(_.left.map(Unauthorized(_)))).andThen(fromEither)
+  )
+
+  def field[K <: Symbol, Repr <: HList](key: Witness.Aux[K])(
+    implicit
+    gen: LabelledGeneric.Aux[AuthInfoT, Repr],
+    select: Selector[Repr, K]
+
+  ): RequestExtractor[select.Out] =
+    apply.map { ai ⇒
+      gen.to(ai)(key)
+    }
+
+  object from extends RecordArgs {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Object name does not match the regular expression '^[A-Z][A-Za-z0-9]*$'](https://www.codacy.com/app/iHeartRadio/asobu/file/2296169980/issues/source?bid=3037259&fileBranchId=3178877#l34)